src/main/java/io/leonis/subra/game/data/Goal.java

```diff
@@ -1,69 +1,37 @@
 package io.leonis.subra.game.data;
 
 import io.leonis.algieba.Spatial;
-import io.leonis.algieba.geometry.CardinalDirection;
-import java.io.Serializable;
+import io.leonis.subra.game.data.Team.TeamIdentity;
 import java.util.Set;
 import lombok.Value;
 import org.nd4j.linalg.api.ndarray.INDArray;
-import org.nd4j.linalg.indexing.NDArrayIndex;
 
 /**
  * The Interface Goal.
  *
- * This interface describes the functionality of a goal.
+ * This interface describes the state of a goal.
  *
- * @author Rimon Oz
  * @author Jeroen de Jong
- * @author Ryan Meulenkamp
  */
-public interface Goal extends Spatial, Serializable {
-  /**
-   * @return The {@link CardinalDirection} to which the back of the goal is facing.
-   */
-  CardinalDirection getCardinalDirection();
-
-  /**
-   * @return The {@link TeamColor color} of the goal.
-   */
-  TeamColor getTeamColor();
+public interface Goal extends GoalDimension.Supplier, Spatial, FieldHalf.Supplier {
 
   /**
-   * @return The width of the goal in mm.
+   * @return The {@link TeamIdentity} defending to this goal.
    */
-  default double getWidth() {
-    return this.getState().getDouble(0, 0);
-  }
-
-  /**
-   * @return The {@link INDArray state vector} of the goal.
-   */
-  INDArray getState();
-
-  /**
-   * @return The depth of the goal in mm.
-   */
-  default double getDepth() {
-    return this.getState().getDouble(1, 0);
-  }
-
-  @Override
-  default INDArray getPosition() {
-    return this.getState().get(NDArrayIndex.interval(2, 4), NDArrayIndex.all());
-  }
+  TeamIdentity getTeamIdentity();
 
   /**
    * @return The X-position coordinate of the {@link Goal}.
    */
   default double getX() {
-    return this.getState().getDouble(2, 0);
+    return this.getPosition().getDouble(2, 0);
```

Comment:
I think this should be `.getDouble(1, 0)`

Reply:
I think it should be `getDouble(0, 0);` Since goalDimension and position-state are now separated.